packages/agreement-email-sender/test/agreementEmailSender.integration.test.ts

```diff
@@ -0,0 +1,283 @@
+import { describe, expect, it, vi } from "vitest";
+import axios from "axios";
+import {
+  AgreementId,
+  DescriptorId,
+  EServiceId,
+  TenantId,
+  UserId,
+  generateId,
+  genericInternalError,
+  toAgreementV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import {
+  getMockAgreement,
+  getMockDescriptor,
+  getMockEService,
+  getMockTenant,
+} from "pagopa-interop-commons-test";
+import { InstitutionResponse } from "pagopa-interop-selfcare-v2-client";
+import { sendAgreementEmail } from "../src/services/agreementEmailSenderService.js";
+import {
+  addOneAgreement,
+  addOneEService,
+  addOneTenant,
+  emailManager,
+  emailManagerConfig,
+  readModelService,
+  selfcareV2ClientMock,
+} from "./utils.js";
+
+describe("agreement email sender", () => {
+  it("should send an email on AgreementSubmitted", async () => {
+    const tenant = { ...getMockTenant(), name: "Tenant" };
+    await addOneTenant(tenant);
+    const agreementId = unsafeBrandId<DescriptorId>(
+      "a8f059b2-3931-4802-ad9d-203936e98c22"
+    );
```

Comment:
This can be removed from each test, no need to generate it separately / hardcode it, `getMockAgreement` already generates a random agreement id

Reply:
I need to hardcode it because it is present in the email.

Reply:
Didn't notice that :) ignore the comment